lib/ElementHandle.js

```diff
@@ -121,9 +119,25 @@ class ElementHandle extends JSHandle {
    * @return {!Promise<?{x: number, y: number, width: number, height: number}>}
    */
   async boundingBox() {
-    const result = await this._client.send('DOM.getBoxModel', {
-      objectId: this._remoteObject.objectId
-    }).catch(error => void debugError(error));
+    let result = null;
+    try {
+      result = await this._client.send('DOM.getBoxModel', {
+        objectId: this._remoteObject.objectId
+      });
+    } catch (protoError) {
+      debugError(protoError);
+
+      const noElementError = await this.executionContext().evaluate(element => {
+        if (!element.isConnected)
+          return 'Node is detached from document';
+        if (element.nodeType !== Node.ELEMENT_NODE)
+          return 'Node is not of type HTMLElement';
+        return null;
+      }, this);
+
+      if (noElementError)
```

Comment:
I think you wanted this code in assertBoundingBox? We intentionally have bounding box throw no errors, and return null for invisible nodes.

Reply:
Fixed